server/db/dbManager.ts

```diff
@@ -55,14 +111,64 @@ export class UserManager {
 		this.collection = collection;
 	}
 
+	async addPostUpvote(postID: string, username: string): Promise<boolean> {
+		const user = await this.collection.findOne({ username });
+		if (user && !Object.prototype.hasOwnProperty.call(user.upvotes, postID)) {
+			const update = { $addFields: {} };
+			// eslint-disable-next-line dot-notation
+			update.$addFields['upvotes'] = { [postID]: 1 };
+			this.collection.updateOne({ username }, [update]);
```

Comment:
In order to reduce code, have validation and error handling, I suggest you to take a look at  `mongoose`. This is Object Data Modeling (ODM) library for MongoDB and it is quite popular. Perhaps, you might want to use it if you like. Tell me want do you think about it? Have you worked with it before?

Reply:
I haven't worked with it before but it seems interesting. Schemas are basically the same thing as the typescript interface I have if I understand correctly. Models though will make code shorter. I think I will switch to it as these update functions are pretty unreadable.